src/AWS.Deploy.CLI/UserInputConfiguration.cs

```diff
@@ -63,10 +69,12 @@ public class UserInputConfiguration<T>
         public bool EmptyOption { get; set; }
 
         public UserInputConfiguration(
+            Func<T, string> idSelector,
             Func<T, string> displaySelector,
             Func<T, bool> defaultSelector,
             string defaultNewName = "")
         {
+            IDSelector = idSelector;
```

Comment:
Not sure why IdSelector is needed. You could keep the logic with target type like in this case Subnet or SecurityGroup and use displaySelector for showing what should be shown to customer.

`AskUserForList` returns a high level object like SeucurityGroup and then mapped to the desired value. This will make the ConsoleUtilities generic than constraining it to string based selection.

Reply:
Given the current restraints of GA, if we decide to add further types to the List functionality, we will need to adapt that in the recipe and support more than `SortedSet<string>` types which will involve us updating the `AskUserForList` to return a generic type.